pkg/controller/job/job_controller.go

```diff
@@ -1071,8 +1088,17 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 		if succeededIndexesStr != jobCtx.job.Status.CompletedIndexes {
 			needsFlush = true
 		}
+
 		jobCtx.job.Status.Succeeded = int32(jobCtx.succeededIndexes.total())
 		jobCtx.job.Status.CompletedIndexes = succeededIndexesStr
+		var failedIndexesStr *string
+		if jobCtx.failedIndexes != nil {
+			failedIndexesStr = pointer.String(jobCtx.failedIndexes.String())
```

Comment:
just to confirm, we are updating these indexes even if we didn't remove the corresponding pod finalizer.

Reply:
If a pod's index is in failedIndexes, then its finalizer removal is not blocked, because failedIndexes are skipped from `podsWithDelayedDeletionPerIndex`. Still, the finalizer removal may fail, and will be retried in the next sync, similarly as we do for succeeded pods.